classes/webservice/ApiNode.php

```diff
@@ -0,0 +1,358 @@
+     * Create new ApiNode instance of type "lang"
+     * Lang ApiNode serves as ApiNode with name. All the translated values are supposed to be children of that Node.
+     *
+     * @param string $name
+     *
+     * @return \ApiNode
+     */
+    private static function lang($name)
+    {
+        return new ApiNode(self::TYPE_LANGUAGE, $name);
+    }
+
+    /**
+     * Create new ApiNode instance of type "parent"
+     * Parent ApiNode serves as ApiNode with name and array of child nodes (and potentionally array of attributes)
+     * Its children nodes are meant to be rendered as associative arrays
+     *
+     * @param string $name
+     * @param array $attributes
+     *
+     * @return \ApiNode
+     */
+    public static function parent($name = null, $attributes = [])
+    {
+        return new ApiNode(self::TYPE_PARENT, $name, null, $attributes);
+    }
+
+    /**
+     * Create new ApiNode instance of type "list"
+     * List ApiNode serves as ApiNode with name and array of child nodes.
+     * Its children nodes are meant to be rendered as non-associative arrays
+     *
+     * @param string $name
+     * @param array $attributes
+     *
+     * @return \ApiNode
+     */
+    public static function list($name = null, $attributes = [])
+    {
+        return new ApiNode(self::TYPE_LIST, $name, null, $attributes);
+    }
+
+    /** @return string */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /** @return string */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /** @return string */
+    public function getValue()
+    {
+        return $this->value;
+    }
+
+    /** @return array|null */
+    public function getAttributes()
+    {
+        return $this->attributes;
+    }
+
+    /** @return array|null */
+    public function getNodes()
+    {
+        return $this->nodes;
+    }
+
+    /**
+     * @param string $type
+     *
+     * @return self
+     */
+    public function setType($type)
+    {
+        $this->type = $type;
+
+        return $this;
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return self
+     */
+    public function setName($name)
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return self
+     */
+    public function setValue($value)
+    {
+        $this->value = $value;
+
+        return $this;
+    }
+
+    /**
+     * @param array $attributes
+     *
+     * @return self
+     */
+    public function setAttributes($attributes)
+    {
+        $this->attributes = $attributes;
+
+        return $this;
+    }
+
+    /**
+     * @param array $nodes
+     *
+     * @return self
+     */
+    public function setNodes($nodes)
+    {
+        $this->nodes = $nodes;
+
+        return $this;
+    }
+
+    /**
+     * @param string $name
+     * @param string $value
+     *
+     * @return self
+     */
+    public function addAttribute($name, $value)
```

Comment:
```suggestion
    public function addAttribute(string $name, string $value): self
```

Reply:
Done in b4d8238 - But i kept `$value` you should be allowed to add simple attribute without value